ironfish/src/rpc/routes/wallet/getNotes.ts

```diff
@@ -48,8 +53,12 @@ router.register<typeof GetAccountNotesStreamRequestSchema, GetAccountNotesStream
       const transaction = await account.getTransaction(transactionHash)
       Assert.isNotUndefined(transaction)
 
+      const asset = await getAssetById(note.assetIdentifier(), node.chain)
```

Comment:
It feels weird to have a utility function which takes the chain as an argument. What do you think about:
1. Moving `getAssetByIdentifier` to a method under the blockchain
2. Checking if the `note.assetIdentifier()` is the native identifier here in the RPC command
3. If (2) is false, call the method in (1)

Reply:
I agree with #1: We shouldn't have this hidden util function it should be `chain.getAssetById` or something like that.

I don't understand the suggestion in #2, what does that mean? We should always call getAssetById() and don't do carvouts.

Reply:
I discussed with Mat about this, I dont like step 2. Currently this is the only place to do the condition check, but as we add more features in wallet for multi assets, there will be more read operation, where we need to do this check. I want to avoid to have the check for any read logic. As a util function, its a uniformed interface for `getAssetByIdentifier` so all the future read can use this as well.

This is the main reason Mat suggested to have the function in util instead under blockchain class.

Reply:
I still don't understand step #2, but It should definitely go in blockchain like all other functions. I don't know what the wallet has to do with this, as they will both have a function like `wallet.getAssetById()` and `chain.getAssetById()` the code won't be shared because they won't even run on the same computer.

Reply:
I will move the function to blockchain.